package/src/main/java/com/microsoft/azure/cosmos/cassandra/CosmosRetryPolicy.java

```diff
@@ -162,4 +168,42 @@ private RetryDecision retryManyTimesWithBackOffOrThrow(int retryNumber) throws I
 
         return retryDecision;
     }
-}
+
+    private RetryDecision retryManyTimesWithMsInfoOrThrow(int retryNumber, String exception) throws InterruptedException {
+
+        RetryDecision retryDecision = null;
+
+        int retryWaitTime = getRetryAfterMs(exception);
+
+        if (this.maxRetryCount == -1) {
+            Thread.sleep(retryWaitTime);
+            retryDecision = RetryDecision.retry(null);
+        } else {
+            System.out.println("retrying after "+retryWaitTime+" millseconds");
+            System.out.println("retryNumber "+retryNumber+" of "+this.maxRetryCount);
+            if (retryNumber < this.maxRetryCount) {
+                Thread.sleep(retryWaitTime);
+                retryDecision = RetryDecision.retry(null);
+            } else {
+                retryDecision = RetryDecision.rethrow();
+            }
+        }
+
+        return retryDecision;
+    }
+
+    public int getRetryAfterMs(String exceptionString){
```

Comment:
We should fallback to exponential backoff if the RetryAfterMS is not present in error message

Reply:
Are we okay with try, catch around the caller of getRetryAfterMs, and only use RetryAfterMs if it is exceptionless?